src/nbl/video/ILogicalDevice.cpp

```diff
@@ -309,312 +309,111 @@ void ILogicalDevice::addCommonShaderDefines(const bool runningInRenderdoc)
+        if (features.indexTypeUint8) addShaderDefineToPool(pool, "NBL_INDEX_TYPE_UINT8");
+        //if (features.deferredHostOperations) addShaderDefineToPool(pool, "NBL_DEFERRED_HOST_OPERATIONS"); // shader doesn't need to know about
+        //if (features.pipelineExecutableInfo) addShaderDefineToPool(pool, "NBL_PIPELINE_EXECUTABLE_INFO"); // shader doesn't need to know about
+        if (features.deviceGeneratedCommands) addShaderDefineToPool(pool, "NBL_DEVICE_GENERATED_COMMANDS");
+        if (features.rayTracingMotionBlur) addShaderDefineToPool(pool, "NBL_RAY_TRACING_MOTION_BLUR");
+        if (features.rayTracingMotionBlurPipelineTraceRaysIndirect) addShaderDefineToPool(pool, "NBL_RAY_TRACING_MOTION_BLUR_PIPELINE_TRACE_RAYS_INDIRECT");
+        if (features.fragmentDensityMapDeferred) addShaderDefineToPool(pool, "NBL_FRAGMENT_DENSITY_MAP_DEFERRED");
+        if (features.rasterizationOrderColorAttachmentAccess) addShaderDefineToPool(pool, "NBL_RASTERIZATION_ORDER_COLOR_ATTACHMENT_ACCESS");
+        if (features.rasterizationOrderDepthAttachmentAccess) addShaderDefineToPool(pool, "NBL_RASTERIZATION_ORDER_DEPTH_ATTACHMENT_ACCESS");
+        if (features.rasterizationOrderStencilAttachmentAccess) addShaderDefineToPool(pool, "NBL_RASTERIZATION_ORDER_STENCIL_ATTACHMENT_ACCESS");
+        if (features.cooperativeMatrixRobustBufferAccess) addShaderDefineToPool(pool, "NBL_COOPERATIVE_MATRIX_ROBUST_BUFFER_ACCESS");
 
-        if (features.geometryShaderPassthrough) addShaderDefineToPool(pool, "NBL_GEOMETRY_SHADER_PASSTHROUGH");
-#endif
 
         if (runningInRenderdoc)
             addShaderDefineToPool(pool,"NBL_RUNNING_IN_RENDERDOC");
 
-#if 0
+
         const auto& limits = m_physicalDevice->getProperties().limits;
-        addShaderDefineToPool(pool,"NBL_MAX_IMAGE_DIMENSION_1D", limits.maxImageDimension1D);
-        addShaderDefineToPool(pool,"NBL_MAX_IMAGE_DIMENSION_2D", limits.maxImageDimension2D);
-        addShaderDefineToPool(pool,"NBL_MAX_IMAGE_DIMENSION_3D", limits.maxImageDimension3D);
-        addShaderDefineToPool(pool,"NBL_MAX_IMAGE_DIMENSION_CUBE", limits.maxImageDimensionCube);
-        addShaderDefineToPool(pool,"NBL_MAX_IMAGE_ARRAY_LAYERS", limits.maxImageArrayLayers);
-        addShaderDefineToPool(pool,"NBL_MAX_BUFFER_VIEW_TEXELS", limits.maxBufferViewTexels);
-        addShaderDefineToPool(pool,"NBL_MAX_UBO_SIZE", limits.maxUBOSize);
-        addShaderDefineToPool(pool,"NBL_MAX_SSBO_SIZE", limits.maxSSBOSize);
-        addShaderDefineToPool(pool,"NBL_MAX_PUSH_CONSTANTS_SIZE", limits.maxPushConstantsSize); // TODO: re-adjust
-        addShaderDefineToPool(pool,"NBL_MAX_MEMORY_ALLOCATION_COUNT", limits.maxMemoryAllocationCount);
-        // addShaderDefineToPool(pool,"NBL_MAX_SAMPLER_ALLOCATION_COUNT",limits.maxSamplerAllocationCount); // shader doesn't need to know about that
-        addShaderDefineToPool(pool,"NBL_BUFFER_IMAGE_GRANULARITY", core::min(limits.bufferImageGranularity,std::numeric_limits<int32_t>::max()));
```

Comment:
look at the code you removed, I need the non-boolean numeric features in the shaders too!!

Every member in `SPhysicalDeviceLimits` needs to be known to the shader.